### PR TITLE
Chore/card add

### PR DESCRIPTION
**Pull request by HubertKwiatkowski**

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -3,31 +3,110 @@ import { getCardComponent, createNewCardComponent } from './ui/card.js';
...
+    cardList.append(addNewCard);
+    const leftButton = addNewCard.querySelector('.btn-left');
+    const rightButton = addNewCard.querySelector('.btn-right');
+
+    const cancelButton = () => {
+      fillCardList();
+      if (!newCardData.front) addNewCard.querySelector('.new-value').value = '';
+    };
+
+    const nextButton = () => {
```
Comment (Sumick) [inline js/app.js]: **nitpick:** I think these methods (`nextButton, backButton, saveButton`) should be placed inside `createNewCardComponent` which creates UI for this card with these methods. Let's split what `addCard` should do (you can think of it as of splitting your code into smaller pieces):

1. It should initialize new card with its handlers using `createNewCardComponent()`
2. It should append this created card to the list
3. It should run `updateCounter`

Basically that's all. Each of the step could be reusable function

Comment (Sumick) [reply js/app.js]: Ahh, now I see that `createNewCardComponent` is your added card component. I suggest you to split these to as 2 different files, so you have your `app.js` clean. Think of it as one file should do one thing.

**Review by Sumick — CHANGES_REQUESTED** [review]
Nice, but we have to improve architecture a little bit. My first long comment is not exactly what I meant there, as I found out that this is different component that you have in `card.js`. We need to improve architecture here, so it will be easier for you to work.

- [x] create `ui/newCard.js` file, and move your `addCard` method (preferably to change method name to `newCard` as well) there. It should be responsible for creating your addable card.
- [ ] rename your method in `ui/card.js` to `card` instead of `createNewCardComponent`, so it's clear what it does
- [ ] You `app.js` should be mostly responsible for initializing required parts of your logic. Don't keep there such long implementations

File: js/ui/newCard.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { getCardList } from './cardList.js';
+import { addCard } from '../data/actions.js';
+import { generateCounter } from './cardCounter.js';
+import { createState } from '../data/createState.js';
+
+let appState = createState();
```
Comment (Sumick) [inline js/ui/newCard.js]: **issue:** You should not initialize `appState` twice - it should be done once somewhere in your `app.js` component. You `UI` files should not know anything about `data` layer

File: js/ui/cardCounter.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export const generateCounter = (appState) => {
```
Comment (Sumick) [inline js/ui/cardCounter.js]: **praise:** Good job on passing state here, as argument - but it will be enough if you just pass `cardsAmount` instead of whole state. You don't need whole state

**Review by Sumick — CHANGES_REQUESTED** [review]
It's starting to look really good! I found three things that we need to fix here:

- [x] You should not call `createState` in your `ui` files. Especially that you did it twice, and it means you have two the same app states to manage
- [x] You should initialize your state somewhere in `app.js`, as one global app state that is shared across your UI components.
- [x] You should parametrize you `UI` components like you did with counter. I mean it should be possible to pass some arguments inside them when calling your UI components functions

File: js/ui/cardList.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { getCardComponent } from './cardComponent.js';
+
+const cardList = document.getElementById('card-list');
```
Comment (Sumick) [inline js/ui/cardList.js]: **nitpick:** This should not be outside of your function. Keep it all encapsulated together

**Review by Sumick — APPROVED** [review]
Much better! Good job. To summarize: try to think of your files in `ui` directory as of React components. They should be re-usable functions that create UI somewhere in your application (depends where you call them)
